assets/javascripts/discourse/components/param-input-form.gjs

```diff
@@ -0,0 +1,328 @@
+        }
+        return null;
+      },
+      int_list: (value) => {
+        return value.split(",").every((i) => /^(-?\d+|null)$/.test(i.trim()))
+          ? null
+          : ERRORS.INVALID;
+      },
+      post_id: (value) => {
+        return isPositiveInt(value) ||
+          /\d+\/\d+(\?u=.*)?$/.test(value) ||
+          /\/t\/[^/]+\/(\d+)(\?u=.*)?/.test(value)
+          ? null
+          : ERRORS.INVALID;
+      },
+      topic_id: (value) => {
+        return isPositiveInt(value) || /\/t\/[^/]+\/(\d+)/.test(value)
+          ? null
+          : ERRORS.INVALID;
+      },
+      category_id: (value) => {
+        return this.site.categoriesById.get(Number(value))
+          ? null
+          : ERRORS.NO_SUCH_CATEGORY;
+      },
+      group_id: (value) => {
+        const groups = this.site.get("groups");
+        if (isPositiveInt(value)) {
+          const intVal = parseInt(value, 10);
+          return groups.find((g) => g.id === intVal)
+            ? null
+            : ERRORS.NO_SUCH_GROUP;
+        } else {
+          return groups.find((g) => g.name === value)
+            ? null
+            : ERRORS.NO_SUCH_GROUP;
+        }
+      },
+    };
+    return VALIDATORS[info.type] ?? (() => null);
+  }
+
+  validatorOf(info) {
+    const getError = this.getErrorFn(info);
+    return (name, value, { addError }) => {
+      // skip require validation for we have used them in @validation
+      if (isEmpty(value) || value == null) {
+        return;
+      }
+      const message = getError(value);
+      if (message != null) {
+        addError(name, { title: info.identifier, message });
+      }
+    };
+  }
+
+  @action
+  async submit() {
+    if (this.form == null) {
+      throw "No form";
+    }
+    await this.form.submit();
+    if (this.serializedData == null) {
+      throw "validation_failed";
+    } else {
+      return this.serializedData;
+    }
+  }
+
+  @action
+  onRegisterApi(form) {
+    this.form = form;
+  }
+
+  @action
+  onSubmit(data) {
+    this.serializedData = null;
+    const serializedData = {};
+    for (const [id, val] of Object.entries(data)) {
+      serializedData[id] =
+        serializeValue(this.infoOf[id].type, val) ?? undefined;
+    }
+    this.serializedData = serializedData;
+  }
+
+  <template>
+    <div class="query-params">
+      <Form
+        @data={{this.data}}
+        @onRegisterApi={{this.onRegisterApi}}
+        @onSubmit={{this.onSubmit}}
+        class="params-form"
+        as |form|
+      >
+        {{#each this.paramInfo as |info|}}
+          <div class="param">
+            <form.Field
+              @name={{info.identifier}}
+              @title={{info.identifier}}
+              @validation={{info.validation}}
+              @validate={{info.validate}}
+              as |field|
+            >
+              <info.component @field={{field}} @info={{info}} />
+            </form.Field>
+          </div>
+        {{/each}}
+      </Form>
```

Comment:
Im impressed, you have greatly managed to integrate form kit into something which is not "standard" at all :clap: